bat-utils/lib/runtime-wallet.js

```diff
@@ -475,4 +496,118 @@ Wallet.providers.mock = {
+    const params = {
+      currency: 'XBT',
+      partner: 'brave',
+      version: '1'
+    }
+    let expires, quote, rate, result
+
+    if (!this.runtime.config.simplex) return
+
+    if (!this.currency.fiatP(currency)) {
+      rate = underscore.pick(this.currency.rates[currency] || {}, fiat)
+      if (!rate) return
+
+      amount = Math.ceil(this.currency.alt2fiat(currency, amount, fiat, true))
+      currency = fiat
+    }
+
+    quote = info.simplex
+    if ((!quote) || (quote.expires <= now)) {
+      result = await braveHapi.wreck.post(this.runtime.config.simplex.url + '/wallet/merchant/v2/quote', {
+        headers: {
+          accept: 'application/json',
+          authorization: 'ApiKey ' + this.runtime.config.simplex.api_key,
+          'content-type': 'application/json'
+        },
+        payload: JSON.stringify({
+          end_user_id: info.paymentId,
+          digital_currency: params.currency,
+          fiat_currency: currency,
+          requested_currency: currency,
+          requested_amount: amount,
+          wallet_id: params.partner
+        })
+      })
+      if (Buffer.isBuffer(result)) try { result = JSON.parse(result) } catch (ex) { result = result.toString() }
+
+      expires = new Date(result.valid_until).getTime()
+      if ((isNaN(expires)) || (!result.quote_id)) return
+
+      quote = underscore.extend(result, { expires: expires })
+    }
+    quote.payment_id = uuid.v4().toLowerCase()
+
+    result = await braveHapi.wreck.post(this.runtime.config.simplex.url + '/wallet/merchant/v1/payments/partner/data', {
+      headers: {
+        accept: 'application/json',
+        authorization: 'ApiKey ' + this.runtime.config.simplex.api_key,
+        'content-type': 'application/json'
+      },
+      payload: JSON.stringify({
+        account_details: {
+          app_provider_id: params.partner,
+          app_version_id: params.version,
+          app_end_user_id: quote.user_id,
+          signup_login: {
+            ip: '213.162.55.5',
+            accept_language: language,
+            http_accept_language: language,
+            timestamp: new Date(now).toISOString()
+          }
+        },
+        transaction_details: {
+          payment_details: {
+            quote_id: quote.quote_id,
+            payment_id: quote.payment_id,
+            order_id: uuid.v4().toLowerCase(),
+            fiat_total_amount: {
+              currency: quote.fiat_money.currency,
+              amount: quote.fiat_money.total_amount
+            },
+            requested_digital_amount: quote.digital_money,
+            destination_wallet: {
+              currency: params.currency,
+              address: info.addresses.BTC
+            }
+          }
+        }
+      })
+    })
+    if (Buffer.isBuffer(result)) try { result = JSON.parse(result) } catch (ex) { result = result.toString() }
+
+// courtesy of https://stackoverflow.com/questions/2076299/how-to-close-current-tab-in-a-browser-window#18776480
```

Comment:
can we remove these comments? they don't seem related

Reply:
i plan to after the browser-side implementation is done. these links are the result of "considerable" research in how to use javascript to close the current tab. i thought it would be as simple as `_self.close()`, but nothing is that simple, sadly.

Reply:
there are other mediums more suited for sharing and storing this research